src/itype.h

```diff
@@ -607,11 +607,18 @@ struct itype {
 
 // Includes food drink and drugs
 struct it_comest : itype {
+    friend class Item_factory;
```

Comment:
Friends are bad, can we just replace this with a constructor? (I'm assuming that's why it's needed.)